Formula/terraform.rb

```diff
@@ -30,7 +30,15 @@ class Terraform < Formula
   fails_with gcc: "5"
 
   def install
-    system "go", "build", *std_go_args(ldflags: "-s -w")
+    ldflags = %w[-s -w]
+    if build.stable?
+      ldflags += %W[
+        -X github.com/hashicorp/terraform/version.Version=#{version}
+        -X github.com/hashicorp/terraform/version.Prerelease=
+      ]
+    end
+
+    system "go", "build", *std_go_args(ldflags: ldflags)
   end
 
   test do
```

Comment:
Let's perhaps add a test to ensure that the right version number is returned – will also help us check if this worked as intended.